src/System.Security.Cryptography.Algorithms/tests/AesCcmTests.cs

```diff
@@ -0,0 +1,433 @@
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using Xunit;
+
+namespace System.Security.Cryptography.Algorithms.Tests
+{
+    public class AesCcmTests
+    {
+        [Fact]
+        public static void EncryptDecryptSimpleNoAAD()
+        {
+            const int datalength = 32;
+            byte[] originalData = Enumerable.Range(1, datalength).Select((x) => (byte)x).ToArray();
+            byte[] nonce = Convert.FromBase64String("tBMp3WSvLDA2ZhtG");
+            byte[] key = Convert.FromBase64String("1aGU7ZDP4Iq+zUaRmXzrLA==");
+
+            byte[] expectedCiphertext = Convert.FromBase64String("WEV/WTngKbmWN8mu8maqdCwzd5F+lsDLXpz/ZKyYNJs=");
+            byte[] expectedTag = Convert.FromBase64String("7Tr31A==");
+
+            byte[] ciphertext = new byte[expectedCiphertext.Length];
+            byte[] tag = new byte[expectedTag.Length];
+
+            using (AesCcm aesCcm = new AesCcm(key))
+            {
+                aesCcm.Encrypt(nonce, originalData, ciphertext, tag);
+                Assert.Equal(expectedCiphertext, ciphertext);
+                Assert.Equal(expectedTag, tag);
+
+                byte[] plaintext = new byte[originalData.Length];
+                aesCcm.Decrypt(nonce, ciphertext, tag, plaintext);
+                Assert.Equal(originalData, plaintext);
+            }
+        }
+
+        [Fact]
+        public static void EncryptDecryptSimpleAAD()
+        {
+            const int datalength = 32;
+            byte[] originalData = Enumerable.Range(1, datalength).Select((x) => (byte)x).ToArray();
+            byte[] additionalData = Enumerable.Range(30, 65).Select((x) => (byte)x).ToArray();
+            byte[] nonce = Convert.FromBase64String("tBMp3WSvLDA2ZhtG");
+            byte[] key = Convert.FromBase64String("1aGU7ZDP4Iq+zUaRmXzrLA==");
+
+            byte[] expectedCiphertext = Convert.FromBase64String("WEV/WTngKbmWN8mu8maqdCwzd5F+lsDLXpz/ZKyYNJs=");
+            byte[] expectedTag = Convert.FromBase64String("j7zpTQ==");
+
+            byte[] ciphertext = new byte[expectedCiphertext.Length];
+            byte[] tag = new byte[expectedTag.Length];
+
+            using (AesCcm aesCcm = new AesCcm(key))
+            {
+                aesCcm.Encrypt(nonce, originalData, ciphertext, tag, additionalData);
+                Assert.Equal(expectedCiphertext, ciphertext);
+                Assert.Equal(expectedTag, tag);
+
+                byte[] plaintext = new byte[originalData.Length];
+                aesCcm.Decrypt(nonce, ciphertext, tag, plaintext, additionalData);
+                Assert.Equal(originalData, plaintext);
+            }
+        }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData(7)]
+        [InlineData(16)]
+        [InlineData(17)]
+        [InlineData(32)]
+        [InlineData(41)]
+        [InlineData(48)]
+        [InlineData(50)]
+        public static void EncryptDecryptRoundtripNoAAD(int dataLength)
```

Comment:
What is this adding over the NIST cases?  (If it's adding something I'm happy to have it, but I'm not seeing what it is)

Reply:
for gcm probably not much - will remove - for ccm - nist test cases are not great (most of them have nonce size we don't support)